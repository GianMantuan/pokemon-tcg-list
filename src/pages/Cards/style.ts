import {Dimensions} from 'react-native';
import styled from 'styled-components/native';

const {height} = Dimensions.get('screen');

export const Container = styled.View`
  margin-top: 20px;
`;
