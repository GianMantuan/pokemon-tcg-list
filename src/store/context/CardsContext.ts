import React from 'react';

const CardsContext = React.createContext({
  cards: [],
});

export default CardsContext;
