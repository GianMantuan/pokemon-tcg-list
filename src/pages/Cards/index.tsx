import { NavigationProp } from '@react-navigation/native';
import React, { useContext, useState } from 'react';
import { useTranslation } from 'react-i18next';
import _ from 'lodash';

import Header from '../../components/Header';
import PokemonCard from '../../components/PokemonCard';
import Search from '../../components/Search';

import { Container } from './style';

import CardsContext from '../../store/context/CardsContext';

type Props = {
  navigation: NavigationProp;
};

export default function Cards({ navigation }: Props): JSX.Element {
  const { t } = useTranslation();
  const { state } = useContext(CardsContext);
  const [filtered, setFiltered] = useState(false);

  function searchFilter(text): void {
    if (text.length >= 3) {
      const formattedText = text.toLowerCase();

      const data = _.filter(state.cards, (pokemon) => {
        return pokemon.name.toLowerCase().includes(formattedText);
      });

      setFiltered(data);
    } else {
      setFiltered(false);
    }
  }

  return (
    <Container>
      <Header
        title="Pokedemon: The Card Game"
        subtitle={t('app_description')}
      />
      <Search searchFunction={searchFilter} />
      <PokemonCard pokemon={filtered || state.cards} navigation={navigation} />
    </Container>
  );
}
