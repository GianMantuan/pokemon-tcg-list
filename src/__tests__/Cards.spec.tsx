import React from 'react'
import { View } from 'react-native'

import Cards from '../pages/Cards/index'
import CardsContext from '../store/context/CardsContext'
import CardsReducer from '../store/reducer/CardsRecuder'

import { cleanup } from '@testing-library/react-native'
import renderer from 'react-test-renderer'

const NavigationProp = (props: Object) => ({
  navigation: {
    navigate: jest.fn()
  },
  ...props
})

describe('Cards Page', () => {
  let navigation: any

  afterEach(cleanup)

  beforeEach(() => {
    navigation = NavigationProp({})
  })

  it('should call Card component with a CardsContext', async () => {
    const state = { cards: [] }

    const cardsComponent = renderer.create(
      <CardsContext.Provider value={{ state, CardsReducer }}>
        <Cards navigation />
      </CardsContext.Provider >
    )

    expect(cardsComponent).toMatchSnapshot();
  })
})
