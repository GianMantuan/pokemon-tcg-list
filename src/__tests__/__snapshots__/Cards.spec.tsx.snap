// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cards Page should call Card component with a CardsContext 1`] = `
<View
  style={
    Array [
      Object {
        "marginTop": 20,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "marginBottom": 20,
          "marginLeft": 20,
          "marginRight": 20,
          "marginTop": 20,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#121212",
            "fontSize": 24,
            "fontWeight": "bold",
            "marginBottom": 10,
          },
        ]
      }
    >
      Pokedemon: The Card Game
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#c1c7c9",
            "fontSize": 18,
            "fontWeight": "400",
          },
        ]
      }
    >
      app_description
    </Text>
  </View>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="search"
    rejectResponderTermination={true}
    style={
      Array [
        Object {
          "backgroundColor": "rgba(230, 230, 230, 0.5)",
          "borderRadius": 14,
          "marginBottom": 20,
          "marginHorizontal": 20,
          "paddingBottom": 15,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 15,
        },
      ]
    }
    underlineColorAndroid="transparent"
  />
  <RCTScrollView
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "minHeight": 1334,
        },
      ]
    }
  >
    <View>
      <RCTScrollView
        activeAnimationOptions={null}
        activeAnimationType="timing"
        activeSlideAlignment="center"
        activeSlideOffset={20}
        apparitionDelay={0}
        automaticallyAdjustContentInsets={false}
        autoplay={false}
        autoplayDelay={1000}
        autoplayInterval={3000}
        callbackOffsetMargin={5}
        containerCustomStyle={Object {}}
        contentContainerCustomStyle={Object {}}
        contentContainerStyle={
          Array [
            Object {
              "paddingLeft": 86.53846153846155,
              "paddingRight": 86.53846153846155,
            },
            Object {},
          ]
        }
        data={Array []}
        decelerationRate="fast"
        directionalLockEnabled={true}
        disableVirtualization={false}
        enableMomentum={false}
        enableSnap={true}
        firstItem={0}
        forwardedRef={[Function]}
        getItem={[Function]}
        getItemCount={[Function]}
        hasParallaxImages={false}
        horizontal={true}
        inactiveSlideOpacity={0.7}
        inactiveSlideScale={0.9}
        inactiveSlideShift={0}
        initialNumToRender={7}
        invertStickyHeaders={false}
        inverted={false}
        itemWidth={576.9230769230769}
        keyExtractor={[Function]}
        layout="default"
        lockScrollTimeoutDuration={1000}
        lockScrollWhileSnapping={false}
        loop={false}
        loopClonesPerSide={3}
        maxToRenderPerBatch={15}
        onContentSizeChange={[Function]}
        onEndReachedThreshold={2}
        onLayout={[Function]}
        onMomentumScrollEnd={[Function]}
        onResponderRelease={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        onStartShouldSetResponderCapture={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        overScrollMode="never"
        pinchGestureEnabled={false}
        removeClippedSubviews={true}
        renderItem={[Function]}
        scrollEnabled={true}
        scrollEventThrottle={1}
        scrollsToTop={false}
        shouldOptimizeUpdates={true}
        showsHorizontalScrollIndicator={false}
        showsVerticalScrollIndicator={false}
        slideStyle={Object {}}
        sliderWidth={750}
        stickyHeaderIndices={Array []}
        style={
          Object {
            "flexDirection": "row",
            "opacity": 0,
            "width": 750,
          }
        }
        swipeThreshold={20}
        updateCellsBatchingPeriod={50}
        useScrollView={false}
        vertical={false}
        viewabilityConfigCallbackPairs={Array []}
        windowSize={15}
      >
        <View />
      </RCTScrollView>
    </View>
  </RCTScrollView>
</View>
`;
